MetaMask/metamask-extension
# Multichain: All Connections Listing

## **Description**
Added a screen to list all connections, including dapps and snaps in one place. Connection Details coming in future PRs.
Clicking "Connections" in the app footer will take you to the All Connections page in fullscreen mode.
Currently, if you have more than 5 total connections (dapps + snaps), it will switch to a tab mode that separates the connections by type.

## **Manual testing steps**

1. Start the extension, and turn on fullscreen mode.

2. Look for the "Connections" option in the app footer. Click on it to navigate to the All Connections page. Verify that the All Connections page opens in fullscreen mode

3. Ensure that all your connections, including dapps and snaps, are listed on this page.

4. If you have less than 5 total connections, try adding more until you have more than 5. This can be done by connecting to new dapps or installing new snaps.

5. Once you have more than 5 total connections, check if the page switches to a tab mode that separates the connections by type (dapps and snaps).

6. Try navigating between the tabs and ensure that the correct connections are displayed for each category.

## **Screenshots/Recordings**

![image](https://github.com/MetaMask/metamask-extension/assets/10986371/10b78570-1b51-4e63-91d7-304540a803f2)

![image](https://github.com/MetaMask/metamask-extension/assets/10986371/7d20ad56-4707-4b53-a8d5-3b43b3d8a002)

## **Related issues**

Fixes https://github.com/MetaMask/MetaMask-planning/issues/916

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
